core/src/main/java/org/mskcc/cbio/portal/scripts/ImportGeneData.java

```diff
@@ -96,13 +96,23 @@ public static void importData(File geneFile, String genomeBuild) throws IOExcept
                     String strAliases = parts[4];
                     String strXrefs = parts[5];
                     String cytoband = parts[7];
-                    String chr = cytoband.split("p|q")[0];
+                    String chr = cytoband.split("p|q|;|c")[0]; // various strange characters were found in this column
                     int referenceGenomeId = DaoReferenceGenome.getReferenceGenomeByBuildName(genomeBuild).getReferenceGenomeId();
                     String desc = parts[8];
                     String type = parts[9];
                     String mainSymbol = parts[10]; // use 10 instead of 2 since column 2 may have duplication
                     Set<String> aliases = new HashSet<String>();
 
+                    // try to get chr from other column if needed
+                    if (chr.equals("-")) {
```

Comment:
combine two ifs into one, so the second condition will not be evaluated if first one is false.
```
       if (chr.equals("-") && (!parts[6].equals("-")) {
              chr = parts[6];
       } else {
              // skip line if still unable to parse chr
             continue;
      }
```

Reply:
Done

Reply:
The unittests revealed a problem with this implementation; the code will go straight to `continue` when `chr` does not equal `"-"`, which is obviously not what we want.

A "synonymous" implementation would be:
```
if (chr.equals("-") && parts[6].equals("-")) {
    continue
} 
if (chr.equals("-")) {
    chr = parts[6];
}
```

However, this requires 2 logical tests while my original implemetation for most of the lines requires 1, thus would be better performance-wise.